Makefile

```diff
@@ -154,7 +154,7 @@ endif
 
 proto-gen: check-proto-deps
 	@echo "Generating Protobuf files"
-	@go run github.com/bufbuild/buf/cmd/buf generate
+	@go run github.com/bufbuild/buf/cmd/buf@v1.29.0 generate
```

Comment:
[Suggestion] To facilitate future refactoring and considering that a specific version of this repo has been designated for use, I recommend introducing a variable to hold this version number and employing it throughout the Makefile in order to simplify version updates in the future.